### ISSUE TITLE
Spack not looking in correct PATH when multiple PATHs used in modulefile

### ISSUE THREAD

**Issue by kellyrowland** [spack/spack#9710]
When pointing Spack to a modulefile that includes multiple instances of `PATH`, the behavior of Spack appears to be to stop at the first instance of `PATH`. This can lead to errors when using Spack to install packages that depend on this modulefile.

Example modulefile contents:

```
/usr/common/software/modulefiles/python/2.7-anaconda-5.2:

conflict	 python 
module-whatis	 Python programming language.
unsetenv	 PYTHONSTARTUP 
setenv		 PYTHONUSERBASE /global/homes/r/rowlandk/.local/cori/2.7-anaconda-5.2 
prepend-path	 PATH /global/common/cori/software/python/2.7-anaconda-5.2/lib/python2.7/site-packages/mpi4py/bin
prepend-path	 PATH /global/common/cori/software/python/2.7-anaconda-5.2/bin 
```
With this, performing a Spack installation with the Python module as a dependency gives:

```
==> Error: RuntimeError: Unable to locate python command in /global/common/cori/software/python/2.7-anaconda-5.2/lib/python2.7/site-packages/mpi4py/bin
```

The second instance of `PATH` in the modulefile is what ends up at the beginning of the user's `PATH` upon loading the module, but the way that Spack reads in the modulefile and takes the first instance of `PATH` leads to this erroneous behavior.

---

We have worked around this issue by changing relevant modulefiles, but I would think that Spack's behavior should read all of the `PATH` in the order resultant from loading the module.

**Comment by adamjstewart** [comment]
I encountered the same problem in #9154.

**Comment by sleak-lbl** [comment]
me too, have added a comment in that issue (#9154)

**Comment by alalazo** [comment]
Duplicate of #9154

**Comment by adamjstewart** [comment]
I'm not sure if I would call this a duplicate. #9154 lists several problems I encountered with bwpy, one of which is module file parsing. It might be good to have a dedicated issue on this.

**Comment by scheibelp** [comment]
> I would think that Spack's behavior should read all of the PATH in the order resultant from loading the module.

If there are multiple PATH entries, is there an easy way to tell which one to use? It appears straightforward to the eye but the naming schemes may be ad-hoc and not map to Spack's own package names in an obvious way.

In #9154 Adam fell back on specifying the external path rather than an external module in `packages.yaml`.

**Comment by kellyrowland** [comment]
Right, this is probably an instance of "conceptually simple enough to a human but tough to explain to a computer".

The easiest implementation that I can think of would be to concatenate the multiple PATH entries and search in all of those directories? That should avoid the issue of whether the modulefile is pre-pending versus appending directories.

**Comment by becker33** [comment]
@kellyrowland can you confirm this is fixed by #12693?
